docs/source/command_line.rst

```diff
@@ -72,6 +72,24 @@ Import discovery
 The following flags customize how exactly mypy discovers and follows
 imports.
 
+``--namespace-packages``
+    This flag enables import discovery to use namespace packages (see
+    `PEP 420`_).  In particular, this allows discovery of imported
+    packages that don't have an ``__init__.py`` (or ``__init__.pyi``)
+    file.
+
+    Namespace packages are found (using the PEP 420 rules, which
+    prefers "classic" packages over namespace packages) along the
+    module search path -- this is primarily set from the source files
+    passed on the command line, the ``MYPYPATH`` environment variable,
+    and the :ref:`mypy_path config option
+    <config-file-import-discovery-global>`.
+
+    Note that this only affects import discovery -- for modules and
+    packages explicitly passed on the command line, mypy still
+    searches for ``__init__.py[i]`` files in order to determine the
+    fully-qualified module/package name.
```

Comment:
Maybe it would be worth talking about what to do in order to explicitly check code in namespace packages?

Reply:
What about things like `mypy -m foo.bar` if `foo` is a namespace package? Users may expect  it to work, since the module name is explicit. Similarly, what about `mypy -p foo` if `foo` is a namespace package?

Reply:
Hm, none of those currently work. Maybe they should, but I don't have time to make it work in this release (command line files/modules/packages use an entirely different code path than imported modules). I'll open a new issue.

Reply:
`mypy -c 'import foo.bar'`, I suppose?

Reply:
Hm, there are still problems with that. I'm not sure why. Will try to look into this before the release goes out.

Reply:
Maybe #5758?